autoload/lsp/ui/vim/completion.vim

```diff
@@ -0,0 +1,240 @@
+" vint: -ProhibitUnusedVariable
+"
+let s:context = {}
+
+function! lsp#ui#vim#completion#setup() abort
+  augroup lsp_ui_vim_completion
+    autocmd!
+    autocmd CompleteDone * call s:on_complete_done()
+  augroup END
+endfunction
+
+"
+" After CompleteDone, v:complete_item's word has been inserted into the line.
+" Yet not inserted commit characters.
+"
+" below example uses # as cursor position.
+"
+" 1. `call getbuf#`<C-x><C-o>
+" 2. select `getbufline` item.
+" 3. Insert commit characters. e.g. `(`
+" 4. fire CompleteDone, then the line is `call getbufline#`
+" 5. call feedkeys to call `s:on_complete_done_after`
+" 6. then the line is `call getbufline(#` in `s:on_complete_done_after`
+"
+function! s:on_complete_done() abort
+  let l:user_data = lsp#omni#extract_user_data_from_completed_item(v:completed_item)
+  if empty(l:user_data)
+    doautocmd User lsp_complete_done
+    return
+  endif
+
+  let s:context.line = getline('.')
```

Comment:
to be consistent with rest of the codebase we should use `let s:context['line'] = getline('.')`

Reply:
fixed